### PR TITLE
T24202

### PR DESCRIPTION
**Pull request by tchx84**
https://phabricator.endlessm.com/T24202

### CODE REVIEW

File: game-state-service [hunk 1]
```diff
@@ -24,12 +27,25 @@ class GameState(object):
...
             self._state = root.get_object()
-            assert self._state is not None
-        except (GLib.Error, AssertionError):
-            traceback.print_exc()
+            if self._state is None:
+                raise InvalidStateError()
+            reset = False
+        except GLib.Error as e:
+            if GLib.quark_from_string(e.domain) == GLib.file_error_quark():
+                print('Can\'t read state from file %s.' % self._FILE_PATH)
```
Comment (ptomato) [inline game-state-service]: Consider using `logging` (https://docs.python.org/3.5/library/logging.html) for these messages. Fabian has a nice example in hack-sound-server.

**Review by ptomato — CHANGES_REQUESTED** [review]
Logic seems fine, but I would suggest using Python's `logging` module.

File: game-state-service (hunk 1 above)
Comment (tchx84) [reply game-state-service]: sure (I was planing to add proper logging support in T24185).

File: game-state-service [hunk 2]
```diff
@@ -24,12 +32,25 @@ class GameState(object):
...
             self._state = root.get_object()
-            assert self._state is not None
-        except (GLib.Error, AssertionError):
-            traceback.print_exc()
+            if self._state is None:
+                raise InvalidStateError()
+            reset = False
+        except GLib.Error as e:
+            if GLib.quark_from_string(e.domain) == GLib.file_error_quark():
+                _logger.error('Can\'t read from file %s' % self._FILE_PATH)
```
Comment (ptomato) [inline game-state-service]: I thought of one more thing: in the logger messages derived from `GLib.Error`s, append `e.message` to the end of the message.

**Review by ptomato — CHANGES_REQUESTED** [review]
Otherwise looks good.

**Review by ptomato — APPROVED** [review]

**Pull request merged**